### PR TITLE
cluster-view: add unit tests, re-design

### PR DESCRIPTION
**Pull request by iusyk**
Redesign:
-VisClient: the way to parse the input nessage is unified, now function getQValue is used instead of getStringValue or getValue. The function returns QVariant, the setter knows which type is expected. -VisClient: array of 'struct PropertyMapper' is used to map property name and setter. Method onTextMessageReceived has a loop to parse the input message and call the corresponded setter to set value.
-Unit tests: new subproject cluster_view_unit_tests is added to support the unit tests The project can be used to work with VisClient undependently on yocto. -Socket and VisClient: now tehre is a base interface to work with socket and simulate the behaviour. The simulation is necessary to support the unit-tests. VisClient has the dedicated constructor to support the customized socket. By defaut - the implementation with QWebSocket is used.

vis-service: redesign, use qt slot-signals to process the message
Redesign has been done to separate the functionality accumulated in
VisClient. Now, it is proposed to use the consumers that get the messges and
send the corresponded commands to teh instrument cluster.
Each consumer corresponds to the cluster command
For example: SpeedConsumer, checks the message if there is an update for speed,
if yes - sends the update to the IC.

### CODE REVIEW

File: meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/ActivatedEvent.cpp
Comment (DKlimenko) [inline meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/ActivatedEvent.cpp:1]: It is better to send unit test data to cluster and see the result on cluster

File: meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/ActivatedEvent.cpp [hunk 1]
```diff
@@ -5,21 +5,27 @@
...
 #include <fcntl.h>
 #include <unistd.h>
+#ifndef UNIT_TEST
 #include <linux/r_taurus_cluster_protocol.h>
+#endif
 #include "ActivatedEvent.h"
 
 ActivatedEvent::ActivatedEvent()
 {
+#ifndef UNIT_TEST
```
Comment (DKlimenko) [inline meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/ActivatedEvent.cpp:15]: It is better to send unit test data to the cluster and see the result on it

File: meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/VisSocket.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+#include <QUrl>
+#include "VisSocket.h"
+
+void VisWebSocket::open(const QUrl &url)
+{
+    onOpen(url);
+}
```
Comment (DKlimenko) [inline meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/VisSocket.cpp:8]: Please add empty line after each function

**Comment by DKlimenko** [comment]
Please check commit message for typo. It would be also helpful to add description how to enable unit tests

File: meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/vis-service/files/Consumers.cpp [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+#include <stdio.h>
+#include <unistd.h>
+#include <stdint.h>
+#include <sys/ioctl.h>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonArray>
+#include <QVector>
+
+#ifndef _TEST_
```
Comment (DKlimenko) [inline meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/vis-service/files/Consumers.cpp:14]: Shouldn't we use UNIT_TEST flag?

File: meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/VisSocket.cpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (DKlimenko) [inline meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/VisSocket.cpp:1]: pls remove empty line

File: meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/ActivatedEvent.cpp (hunk 1 above)
Comment (iusyk) [reply meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/cluster-view/files/ActivatedEvent.cpp:15]: it has to be something like : test-mode with the predefined route - will be done

File: meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/vis-service/files/Consumers.cpp (hunk 3 above)
Comment (iusyk) [reply meta-xt-prod-cockpit-rcar-driver-domain/recipes-extended/vis-service/files/Consumers.cpp:15]: no. here is not a unit tests

**Comment by iusyk** [comment]
not relevant for the project

**Pull request closed without merging**